hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactory.kt

```diff
@@ -79,11 +79,8 @@ class HmppsQueueFactory(
     .errorHandler(
       object : ErrorHandler<Any> {
         override fun handle(message: org.springframework.messaging.Message<Any>, t: Throwable) {
-          // SDI-477 remove this logging when we are comfortable that all is working as expected - instant retries
-          log.info("Setting visibility of messageId ${message.headers["id"]} to $errorVisibilityTimeout (to initiate faster retry) after receiving exception ${t.cause?.cause?.cause}")
-          val sqsVisibility = message.headers["Sqs_VisibilityTimeout"] as QueueMessageVisibility
-          sqsVisibility.changeTo(errorVisibilityTimeout)
-          throw t
+           errorVisibilityHandler.setErrorVisibilityTimeout(message, queueId)
+           throw t
```

Comment:
It would have been nice to make this error handler injectable so that teams could override the default error handler. Unfortunately we need the queueId so we couldn't do that.